src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -875,18 +875,6 @@ HeapWord* ShenandoahHeap::allocate_memory(ShenandoahAllocRequest& req) {
       result = allocate_memory_under_lock(req, in_new_region);
     }
 
-    // Check that gc overhead is not exceeded.
-    //
-    // Shenandoah will grind along for quite a while allocating one
-    // object at a time using shared (non-tlab) allocations. This check
-    // is testing that the GC overhead limit has not been exceeded.
-    // This will notify the collector to start a cycle, but will raise
-    // an OOME to the mutator if the last Full GCs have not made progress.
-    if (result == nullptr && !req.is_lab_alloc() && get_gc_no_progress_count() > ShenandoahNoProgressThreshold) {
```

Comment:
We need to be careful here. This essentially backs out the fix for:
* https://bugs.openjdk.org/browse/JDK-8316632
* https://bugs.openjdk.org/browse/JDK-8314935

For the original change, I ran tier1, tier2 and tier3 tests with Shenandoah enabled on x86 and aarch64, per @shipilev 's request (https://github.com/openjdk/jdk/pull/15852#issuecomment-1729099298).
```
make test TEST="tier1 tier2 tier3" TEST_VM_OPTS="-XX:+UseShenandoahGC" CONF=fastdebug
```
These tests, specifically,  will likely timeout without this check:
```
make test TEST="runtime/reflect/ReflectOutOfMemoryError.java gc/InfiniteList.java runtime/ClassInitErrors/TestOutOfMemoryDuringInit.java compiler/uncommontrap/TestDeoptOOM.java" TEST_VM_OPTS="-Xlog:gc*=info:file=/tmp/oome.log -XX:+UseShenandoahGC" JTREG="REPEAT_COUNT=50" CONF=fastdebug
```

Reply:
I'll run these tests and report the results.

Reply:
For the four explicit tests, 50 runs each, all passed.

Reply:
Am wondering if there might have been some other compounding issue, since resolved, that contributed to the original issue.